cmd/source_profile_test.go

```diff
@@ -0,0 +1,55 @@
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewSourceProfileFile(t *testing.T) {
+	testCases := []struct {
+		name         string
+		params       map[string]string
+		pipedToStdin bool
+		want         SourceProfileFile
+	}{
+		{
+			name:         "no params, file piped",
+			params:       map[string]string{},
+			pipedToStdin: true,
+			want:         SourceProfileFile{format: "dump"},
+		},
+		{
+			name:         "format param, file piped",
+			params:       map[string]string{"format": "dump"},
+			pipedToStdin: true,
+			want:         SourceProfileFile{format: "dump"},
+		},
+		{
+			name:         "format and path param, file piped -- piped file takes precedence",
+			params:       map[string]string{"format": "dump", "file": "file1.mysqldump"},
+			pipedToStdin: true,
+			want:         SourceProfileFile{format: "dump"},
+		},
+		{
+			name:         "format and path param, no file piped",
+			params:       map[string]string{"format": "dump", "file": "file1.mysqldump"},
+			pipedToStdin: false,
+			want:         SourceProfileFile{format: "dump", path: "file1.mysqldump"},
```

Comment:
For the case when format dump is specified, no file is specified, piped is False, will we throw an error? I may be wrong but as per my understanding, it will default to direct connect currently.

Reply:
This is an interesting corner case, and I suspect there may be many more such where users pass in invalid combination of flags and source-profile key-value pairs. I agree that we should catch as many of these invalid conditions as possible and return appropriate errors, but curious to know @bharadwaj-aditya's opinion as well before I make this change.

Reply:
I now consider presence of either/both of "file" or "format" as the user's intention to use a file input. Have added corresponding error handling.

Reply:
this handling sounds reasonable to me